lib/openstack_query/handlers/handler_base.py

```diff
@@ -0,0 +1,19 @@
+from enum import Enum
+from abc import ABC, abstractmethod
+from typing import Callable
+from enums.query.query_presets import QueryPresets
+
+
+class HandlerBase(ABC):
+    """
+    Abstract Base class for handlers.
+    This class is an abstract class for a generic handler
+    A handler stores a mapping between a preset and a function.
+    """
+
+    def _check_supported(self, preset: QueryPresets, prop: Enum) -> bool:
```

Comment:
I think this should be public, rather than private since the single/both deriving classes implement the public version of this